.github/workflows/ci.yml

```diff
@@ -1,102 +1,72 @@
+      # `snok/install-poetry` seems to be minimal yet complete, and really just calls out to
+      # Poetry's default install script, which feels correct. I pin the Poetry version here
+      # because Poetry does occasionally change APIs between versions and I don't want my
+      # actions to break if it does.
+      #
+      # The key configuration value here is `virtualenvs-in-project: true`: this creates the
+      # venv as a `.venv` in your testing directory, which allows the next step to easily
+      # cache it.
+      - uses: snok/install-poetry@v1
         with:
-          commit_message: Auto committing tables
-          # Optional commit user and author settings
-          # commit_user_name: My GitHub Actions Bot # defaults to "github-actions[bot]"
-          # commit_user_email: my-github-actions-bot@example.org # defaults to "github-actions[bot]@users.noreply.github.com"
-          # commit_author: Author <actions@github.com> # defaults to author of the commit that triggered the run
-          file_pattern: out
-          add_options: '-f'
-          skip_dirty_check: true
-
-      # - name: Report
-      #   run: |
-      #     make report-coverage
-
-      # - name: Upload code coverage result to CodeCov
-      #   uses: codecov/codecov-action@v2
-      #   with:
-      #     token: ${{ secrets.CODECOV_TOKEN }} 
-      #     files: coverage.xml
-      #     fail_ci_if_error: true
-      #     verbose: true 
+          version: 1.1.12
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      # Cache your dependencies (i.e. all the stuff in your `pyproject.toml`). Note the cache
+      # key: if you're using multiple Python versions, or multiple OSes, you'd need to include
+      # them in the cache key. I'm not, so it can be simple and just depend on the poetry.lock.
+      - name: cache deps
+        id: cache-deps
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: pydeps-${{ hashFiles('**/poetry.lock') }}
+
+      # Install dependencies. `--no-root` means "install all dependencies but not the project
+      # itself", which is what you want to avoid caching _your_ code. The `if` statement
+      # ensures this only runs on a cache miss.
+      - run: poetry install --no-interaction --no-root
+        if: steps.cache-deps.outputs.cache-hit != 'true'
+
+      # Now install _your_ project. This isn't necessary for many types of projects -- particularly
+      # things like Django apps don't need this. But it's a good idea since it fully-exercises the
+      # pyproject.toml and makes that if you add things like console-scripts at some point that
+      # they'll be installed and working.
+      - run: poetry install --no-interaction
+
+      # And finally run tests. I'm using pytest and all my pytest config is in my `pyproject.toml`
+      # so this line is super-simple. But it could be as complex as you need.
+      - run: poetry run pytest
```

Comment:
```suggestion
      - run: make test
```

Reply:
I think we should consider getting rid of the makefile.  We have Python commands in `main.py` that can be run from the command line via the `click` library.  We're also defining Makefile commands to run the Python commands - it seems circular.  I'm open to changing my mind here, but seems like we're using both `click` and the Makefile to expose a command line interface to interact with this lib.

Reply:
Well those are two different things. The Makefile defines the common CLI invocations used in the repo. It's actually quick useful for newcomers. The includes our scripts, but also the various test and lint scripts (which are unrelated to `click`).

The `click` stuff makes our script invokable by CLI, allowing us to include those invocations in the Makefile, along with all the other commands.

I think that setup makes a lot of sense to me.

Reply:
@wjones127 - thanks for clarifying.  Made your suggested change.